File: testCode/js/WebGL.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
import Common_1 from "./Common.js";
import Output_1 from "./Output.js";
import Mouse_1 from "./Mouse.js";
var WebGL = /** @class */ (function () {
    class WebGL {
        constructor(_a) {
            var $wrapper = _a.$wrapper;
            this.$wrapper = $wrapper;
            Common_1.init();
            Mouse_1.init();
            // this.init();
            this.$wrapper.prepend(Common_1.renderer.domElement);
            this.output = new Output_1();
            this.loop();
            window.addEventListener("resize", this.resize.bind(this));
        }
        resize() {
            Common_1.resize();
            this.output.resize();
        }
        render() {
            Mouse_1.update();
            Common_1.update();
            this.output.update();
        }
        loop() {
            this.render();
            requestAnimationFrame(this.loop.bind(this));
        }
    }
    return WebGL;
}());
const _default = WebGL;
export { _default as default };

File: src/WebGL.lua
local Common = require("src.Common")
local Output = require("src.Output")
local Mouse = require("src.Mouse")

local window = js.global

return function(properties)
    local self = {}
    self.properties = properties

    Common.init()
    Mouse.init()

    self.init()
    self.loop()

    window:addEventListener("resize", self.resize)

    self.init = function()
        self.properties.wrapper.prepend(Common.renderer.domElement)
        self.output = Output()
    end

    self.resize = function()
        Common.resize()
        self.output.resize()
    end

    self.render = function()
        Mouse.update()
        Common.update()
        self.output.update()
    end

    self.loop = function()
        self.render()
        window:requestAnimationFrame(self.loop)
    end
end